src/components/molecules/ExportProgressItem.tsx

```diff
@@ -0,0 +1,615 @@
+      includeFrontMatter,
+      addExportError,
+    ]
+  )
+
+  const exportDocuments = useCallback(async () => {
+    const savePathname = await openDialog()
+    if (!savePathname) {
+      return
+    }
+
+    const storage = storageMap[storageId]
+    if (!storage) {
+      pushMessage({
+        title: 'Cannot find storage',
+        description: 'Please check storage and try again later.',
+      })
+      console.warn('Storage ID cannot be found: ' + storageId)
+      return
+    }
+
+    setShowExportProgress(true)
+    try {
+      const rootDir = exportSettings.exportingStorage
+        ? join(savePathname, storage.name)
+        : savePathname
+      const allDocsMap: AllDocsMap = await storage.db.getAllDocsMap()
+      const { noteMap, folderMap } = allDocsMap
+
+      updateExportState({ exportOperation: 'Creating directories' })
+      try {
+        if (exportSettings.recursive) {
+          await createExportDirectories(folderMap, rootDir)
+        } else {
+          await mkdir(join(rootDir, exportSettings.folderName))
+        }
+      } catch (err) {
+        addExportError('Cannot make directories: ' + err)
+        console.warn('Please check write access for destination directories: ')
+      }
+
+      if (exportErrors.length === 0) {
+        updateExportState({ exportOperation: 'Exporting notes' })
+
+        try {
+          const exportedNotesCount = await exportNotes(
+            noteMap,
+            storage.attachmentMap,
+            rootDir
+          )
+          if (isExportingCanceled()) {
+            pushMessage({
+              title: 'Export canceled',
+              description: `Exported ${exportedNotesCount} notes.`,
+            })
+          } else {
+            pushMessage({
+              title: 'Export finished',
+              description: `Exported ${exportedNotesCount} notes successfully.`,
+            })
+          }
+        } catch (err) {
+          addExportError('Exporting notes error: ' + err)
+          console.warn('Unknown failure while exporting notes, reason: ' + err)
+        }
+      }
+
+      if (exportErrors.length > 0) {
+        pushMessage({
+          title: 'Export errors',
+          description: `Status: [${exportErrors.join(', ')}].`,
+        })
+        console.warn('Errors during export: ', exportErrors)
+      }
+    } catch (error) {
+      pushMessage({
+        title: 'Export failed',
+        description: 'Reason: ' + error,
+      })
+      console.warn('Export failed with: ', error)
+      console.warn('Errors during export: ', exportErrors)
+    }
+  }, [
+    openDialog,
+    storageMap,
+    storageId,
+    pushMessage,
+    exportSettings.exportingStorage,
+    exportSettings.recursive,
+    exportSettings.folderName,
+    updateExportState,
+    exportErrors,
+    isExportingCanceled,
+    createExportDirectories,
+    addExportError,
+    exportNotes,
+  ])
+
+  useEffectOnce(() => {
+    exportDocuments().then(onFinish).catch(onFinish)
```

Comment:
If we run `onFinish` either results, please use `.finally` method. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
Refactored to finally!